lib/Importer/ProtobufLoader.cpp

```diff
@@ -37,34 +37,58 @@ Tensor *ProtobufLoader::getTensorByName(const std::string &name) {
   return tensors_[name];
 }
 
-Node *ProtobufLoader::getNodeByName(const std::string &name) {
+Node *ProtobufLoader::getNodeByNameOrNull(const std::string &name) const {
   auto it = nodeByName_.find(name);
   if (it != nodeByName_.end()) {
     return it->second;
   }
 
-  llvm_unreachable("Could not find a node with this name.");
   return nullptr;
 }
 
+Node *ProtobufLoader::getNodeByName(const std::string &name) const {
+  assert(hasNodeByName(name) && "No node under that name");
+  auto *node = getNodeByNameOrNull(name);
+  assert(node && "Null is under that name??");
+  return node;
+}
+
+Node *ProtobufLoader::createNodeImpl(const std::string &name, Tensor &tensor,
+                                     VisibilityKind visibilityKind,
+                                     Variable::TrainKind trainKind) {
+  auto *V = G_.getParent()->createVariable(
+      tensor.getElementType(), tensor.dims(), name, visibilityKind, trainKind);
+  V->copyFrom(&tensor);
+  return V;
+}
+
+Node *ProtobufLoader::createNode(const std::string &name, Tensor &tensor) {
+  return createNodeImpl(name, tensor, VisibilityKind::Private,
```

Comment:
I'm not sure if having separate `createNodeImpl` makes things any better. I think it just adds more code, comparing to previous version.

Reply:
I don't have a strong opinion either ways.
Should I go back to only one createNode with the default parameters?

Reply:
Well, you still have Variable's defaults hardcoded here in `createNode()`. Given that we don't have solution not to duplicate them (and also that we still have defaults in `createAndRememberNode()`), it's better to have 2 methods rather than 3.

Reply:
Good point!
Updated.